test/galleryDom.test.js

```diff
@@ -48,3 +48,63 @@ describe('Gallery Counter', () => {
     expect(message).toBe('Count: 0')
   })
 })
+
+// Mock fetch globally if using Jest
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    json: () => Promise.resolve({
+      payload: {
+        pets: [
+          {
+            type: 'dog',
+            gender: 'male',
+            rating: 5,
+            img: 'dog.jpg'
+          }
+        ]
+      }
+    })
+  })
+)
+
+describe('json file fetching data on gallery.html', () => {
+  beforeEach(() => {
+    // Reset fetch mock before each test
+    fetch.mockClear()
+  })
+
+  it('Populate galleryContainer with pet elements from JSON', async () => {
+    const dom = new JSDOM('<!DOCTYPE html><html><body><div id="img-container"></div></body></html>')
+    global.document = dom.window.document
+
+    await getPets('#img-container')
+
+    const galleryContainer = document.querySelector('#img-container')
+    const galleryItems = galleryContainer.querySelectorAll('.gallery-items')
+
+    // Check if the gallery items are correctly added to the galleryContainer
+    expect(galleryItems.length).toBeGreaterThan(0)
```

Comment:
The length depends on the json file right? 
If there are 5 items there, the length should be 5.

If the mocked json contains 1 element and the gallery function shows 10 times the same element, the test will pass but the behaviour is broken